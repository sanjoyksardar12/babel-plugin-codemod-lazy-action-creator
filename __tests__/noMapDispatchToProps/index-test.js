const babel = require("@babel/core");
const plugin = require("../../index.js");

const babelConfig = { presets: ["@babel/preset-react"], plugins: [plugin] };

var noMapDispatToPropParam = `
  import React, { Component } from "react";
  import { connect } from "react-redux";
  class App extends Component {
    render() {
      const { addition, subtraction, multiplication } = this.props;
      return (
        <div className="App">
          <h1>Map dispatchToProps as object</h1>
          <h2> Multiplication: {multiplication} </h2>
          <h2> Addition: {addition} </h2>
          <h2> Subtraction: {subtraction} </h2>
        </div>
      );
    }
  }

  const mapStateToProps = state=>{
    return{
      ...state
    }
  }
  export default connect(mapStateToProps)(App);
`;

it("no MapDispatToProp Param", () => {
  const { code } = babel.transform(noMapDispatToPropParam, babelConfig);
  expect(code).toMatchSnapshot();
});


var mapDispatchToPropsAsUndified = `
  import React, { Component } from "react";
  import { connect } from "react-redux";
  class App extends Component {
    render() {
      const { addition, subtraction, multiplication } = this.props;
      return (
        <div className="App">
          <h1>Map dispatchToProps as object</h1>
          <h2> Multiplication: {multiplication} </h2>
          <h2> Addition: {addition} </h2>
          <h2> Subtraction: {subtraction} </h2>
        </div>
      );
    }
  }

  const mapStateToProps = state=>{
    return{
      ...state
    }
  }
  export default connect(mapStateToProps, undefined)(App);
`;

it("MapDispatToProp as undefined", () => {
  const { code } = babel.transform(mapDispatchToPropsAsUndified, babelConfig);
  expect(code).toMatchSnapshot()
});


var mapDispatchToPropsAsNull = `
  import React, { Component } from "react";
  import { connect } from "react-redux";
  class App extends Component {
    render() {
      const { addition, subtraction, multiplication } = this.props;
      return (
        <div className="App">
          <h1>Map dispatchToProps as object</h1>
          <h2> Multiplication: {multiplication} </h2>
          <h2> Addition: {addition} </h2>
          <h2> Subtraction: {subtraction} </h2>
        </div>
      );
    }
  }

  const mapStateToProps = state=>{
    return{
      ...state
    }
  }
  export default connect(mapStateToProps, null)(App);
`;

it("MapDispatToProp as null", () => {
  const { code } = babel.transform(mapDispatchToPropsAsNull, babelConfig);
  expect(code).toMatchSnapshot()
})





var mapDispatchToPropsAsEmptyObj = `
  import React, { Component } from "react";
  import { connect } from "react-redux";
  class App extends Component {
    render() {
      const { addition, subtraction, multiplication } = this.props;
      return (
        <div className="App">
          <h1>Map dispatchToProps as object</h1>
          <h2> Multiplication: {multiplication} </h2>
          <h2> Addition: {addition} </h2>
          <h2> Subtraction: {subtraction} </h2>
        </div>
      );
    }
  }

  const mapStateToProps = state=>{
    return{
      ...state
    }
  }
  export default connect(mapStateToProps, {})(App);
`;

it("MapDispatToProp as empty obj", () => {
  const { code } = babel.transform(mapDispatchToPropsAsEmptyObj, babelConfig);
  expect(code).toMatchSnapshot()
})



