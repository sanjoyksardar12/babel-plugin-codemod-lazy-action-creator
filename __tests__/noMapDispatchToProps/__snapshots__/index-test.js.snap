// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapDispatToProp as empty obj 1`] = `
"import React, { Component } from \\"react\\";
import { connect } from \\"react-redux\\";

class App extends Component {
  render() {
    const {
      addition,
      subtraction,
      multiplication
    } = this.props;
    return /*#__PURE__*/React.createElement(\\"div\\", {
      className: \\"App\\"
    }, /*#__PURE__*/React.createElement(\\"h1\\", null, \\"Map dispatchToProps as object\\"), /*#__PURE__*/React.createElement(\\"h2\\", null, \\" Multiplication: \\", multiplication, \\" \\"), /*#__PURE__*/React.createElement(\\"h2\\", null, \\" Addition: \\", addition, \\" \\"), /*#__PURE__*/React.createElement(\\"h2\\", null, \\" Subtraction: \\", subtraction, \\" \\"));
  }

}

const mapStateToProps = state => {
  return { ...state
  };
};

export default connect(mapStateToProps, dispatch => ({}))(App);"
`;

exports[`MapDispatToProp as null 1`] = `
"import React, { Component } from \\"react\\";
import { connect } from \\"react-redux\\";

class App extends Component {
  render() {
    const {
      addition,
      subtraction,
      multiplication
    } = this.props;
    return /*#__PURE__*/React.createElement(\\"div\\", {
      className: \\"App\\"
    }, /*#__PURE__*/React.createElement(\\"h1\\", null, \\"Map dispatchToProps as object\\"), /*#__PURE__*/React.createElement(\\"h2\\", null, \\" Multiplication: \\", multiplication, \\" \\"), /*#__PURE__*/React.createElement(\\"h2\\", null, \\" Addition: \\", addition, \\" \\"), /*#__PURE__*/React.createElement(\\"h2\\", null, \\" Subtraction: \\", subtraction, \\" \\"));
  }

}

const mapStateToProps = state => {
  return { ...state
  };
};

export default connect(mapStateToProps, null)(App);"
`;

exports[`MapDispatToProp as undefined 1`] = `
"import React, { Component } from \\"react\\";
import { connect } from \\"react-redux\\";

class App extends Component {
  render() {
    const {
      addition,
      subtraction,
      multiplication
    } = this.props;
    return /*#__PURE__*/React.createElement(\\"div\\", {
      className: \\"App\\"
    }, /*#__PURE__*/React.createElement(\\"h1\\", null, \\"Map dispatchToProps as object\\"), /*#__PURE__*/React.createElement(\\"h2\\", null, \\" Multiplication: \\", multiplication, \\" \\"), /*#__PURE__*/React.createElement(\\"h2\\", null, \\" Addition: \\", addition, \\" \\"), /*#__PURE__*/React.createElement(\\"h2\\", null, \\" Subtraction: \\", subtraction, \\" \\"));
  }

}

const mapStateToProps = state => {
  return { ...state
  };
};

export default connect(mapStateToProps, undefined)(App);"
`;

exports[`no MapDispatToProp Param 1`] = `
"import React, { Component } from \\"react\\";
import { connect } from \\"react-redux\\";

class App extends Component {
  render() {
    const {
      addition,
      subtraction,
      multiplication
    } = this.props;
    return /*#__PURE__*/React.createElement(\\"div\\", {
      className: \\"App\\"
    }, /*#__PURE__*/React.createElement(\\"h1\\", null, \\"Map dispatchToProps as object\\"), /*#__PURE__*/React.createElement(\\"h2\\", null, \\" Multiplication: \\", multiplication, \\" \\"), /*#__PURE__*/React.createElement(\\"h2\\", null, \\" Addition: \\", addition, \\" \\"), /*#__PURE__*/React.createElement(\\"h2\\", null, \\" Subtraction: \\", subtraction, \\" \\"));
  }

}

const mapStateToProps = state => {
  return { ...state
  };
};

export default connect(mapStateToProps)(App);"
`;
